app/gui/src/dashboard/typings.d.ts

```diff
@@ -0,0 +1,16 @@
+/** @file Module augmentations. */
+import type { ReactNode, Ref, RefAttributes } from 'react'
+
+declare module 'react' {
+  function forwardRef<T, P = object>(
+    render: (props: P, ref: Ref<T>) => ReactNode | null,
+  ): (props: P & RefAttributes<T>) => ReactNode | null
+
+  // Technically incorrect, as internally it is a `MemoExoticComponent`,
+  // however in practice it should be indistinguishable from the original component.
+  function memo<P = object>(
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Component: (props: P) => ReactNode | null,
```

Comment:
IIRC ReactNode already includes 'null'

Also I'm not sure that number/false are valid return types for components.

But if it matches the react types then fine

Reply:
yeah :face_with_spiral_eyes: it seems like it does match react types. who knows